### ISSUE TITLE
statsmodels failing to build with numpy-1.10.0.post2

### ISSUE THREAD

**Issue by g2boojum** [statsmodels/statsmodels#2645]
```python
Traceback (most recent call last):
  File "setup.py", line 401, in <module>
    install_requires) = check_dependency_versions(min_versions)
  File "setup.py", line 119, in check_dependency_versions
    if not (StrictVersion(strip_rc(npversion)) >= min_versions['numpy']):
  File "/usr/lib64/python3.5/distutils/version.py", line 40, in __init__
    self.parse(vstring)
  File "/usr/lib64/python3.5/distutils/version.py", line 137, in parse
    raise ValueError("invalid version number '%s'" % vstring)
ValueError: invalid version number '1.10.0.post2'
```

**Comment by josef-pkt** [comment]
Is '1.10.0.post2' supposed to be a valid version number?

I remember we updated our own version number to be correct for recent definitions, but I don't think we reviewed our version checks.

**Comment by saketkc** [comment]
`1.10.0.post2` is a valid version under [PEP-0440] https://www.python.org/dev/peps/pep-0440/
I am not sure if `distutils` has support for this(atleast not `StrictVersion` currently in use in `statsmodels`). [packaging](https://pypi.python.org/pypi/packaging) supports both `PEP-0440` and LegacyVersion checks, but would add up as a build time dependency

**Comment by josef-pkt** [comment]
Thanks Saket for checking.

added prio-high, I guess we can specifically strip it like rc, but I'm not sure what to do about statsmodels 0.6. 
0.7 also still requires attention before a release.

@g2boojum  Thanks for the report, I haven't looked at the new numpy yet.

**Comment by g2boojum** [comment]
@josef-pkt My pleasure.  I worked around it by building numpy HEAD, which turned out to be version 1.11.0.dev0+d573c63.  The statsmodels setup.py handled that version just fine.

Thanks for the quick response!  Statsmodels makes it so I don't have to use R, and I'm extremely appreciative.  (I'm not opposed to R, but I much prefer Python.)

**Comment by dudalev** [comment]
Apparently, numpy itself ran into the (almost) the same problem - see numpy/numpy#6431

**Comment by josef-pkt** [comment]
Based on the discussion on the numpy mailing list and the numpy issue, I expect that this will be resolved from the numpy side.
We still fix it to make the version check more robust in the future, but the urgency is gone.

**Comment by jensenbox** [comment]
Is Numpy 1.9.1 the latest version to pin pip to in the mean time?

**Comment by josef-pkt** [comment]
On pypi the latest version without `post` is 1.9.3 https://pypi.python.org/simple/numpy/
Anaconda seems to have 1.10.0 without post https://anaconda.org/anaconda/numpy/files
sourceforge also has 1.10.0 without `post` http://sourceforge.net/projects/numpy/files/NumPy/1.10.0/

Also the problem with the version check in in setup.py is install time.
But I just checked and we use `from .scipy import NumpyVersion` for runtime checks, which I guess is also broken based on the comments by Ralf in the numpy github issue.
`statsmodels.compat.scipy.NumpyVersion` is our copy of the same function in scipy or numpy.

**Comment by rgommers** [comment]
Yeah. This should be solved by a 1.10.1 release asap, and never using ``.postN`` again (which has already been decided). And a regexp check on version number will be put into numpy itself to not have this happen again.

**Comment by bashtage** [comment]
@josef-pkt This seems like it has self resolved.
